Chandan-devs-tech/urban-seller
# Add Frontend Implementation with React Components and Routing

# Description

## This pull request includes the implementation of the front end of the project using React. The changes include setting up the initial structure, adding essential components, and configuring routing. The following tasks were completed:

- React Setup & Git Initialize
  - Initialized the React project.
  - Configured Git for version control.
 
- Header & Footer Components
  - Created reusable Header and Footer components.

- Custom Bootstrap Theme
  - Applied a custom Bootstrap theme for styling.

- List Products
  - Implemented a component to list products.
  - Added sample product data and images.

- Implement React Router
  - Configured React Router for navigation between different pages.

- Rating Component
  - Created a reusable Rating component to display product ratings.

- Product Details Page
  - Developed a detailed view of individual products.

- Files Added:

  - React Setup & Git Initialize
  - React Setup & Git Initialize_en.srt
  - Header & Footer Components
  - Header & Footer Componenets_en.srt
  - Custom Bootstrap Theme
  - Custom Bootstrap Theme_en.srt
  - List Products
  - List Products_en.srt
  - products-and-images/
  - Implement React Router
  - Implement React Router_en.srt
  - Rating Component
  - Rating Component_en.srt
  - Product Details Page
  - Product Details Page_en.srt

- Summary of Changes:

  - Initial setup and configuration of React and Git.
  - Development of Header and Footer components.
  - Application of a custom Bootstrap theme.
  - Implementation of product listing with images.
  - Integration of React Router for handling different routes.
  - Creation of a reusable rating component.
  - Development of a product details page for detailed information.

- How to Test:

  - Clone the repository and switch to the frontend branch.
  - Run npm install to install dependencies.
  - Run npm start to launch the development server.
  - Navigate through the application to ensure all components and routes are working correctly.
